### PR TITLE
Resolve base docker images correctly for .NET 5.0+ projects

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Fixes https://github.com/dotnet/tye/issues/683. Note that the naming scheme is dependent on the TFM of the project: https://hub.docker.com/_/microsoft-dotnet-core

### CODE REVIEW

File: src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs [hunk 1]
```diff
@@ -123,19 +123,38 @@ private async Task TransformProjectToContainer(Service service, ProjectRunInfo p
...
-                baseImage = project.IsAspNet ? "mcr.microsoft.com/dotnet/core/aspnet" : "mcr.microsoft.com/dotnet/core/runtime";
-            }
+                Version? baseImageVersion = null;
+                try
+                {
+                    baseImageVersion = new Version(baseImageTag);
+                }
+                catch
+                {
+                    _logger.LogInformation($"Could not determine version of docker base image for tag: {baseImageTag}");
```
Comment (jkotalik) [inline src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs:144]: Shouldn't you still let this throw?

**Review by jkotalik — APPROVED** [review]

File: src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs (hunk 1 above)
Comment (JunTaoLuo) [reply src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs:144]: Curious if we should, currently it falls back to 2.1/3.1 but I dunno if that makes sense. If we throw we should probably advise selecting an explicit base image via the option.

Comment (JunTaoLuo) [reply src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs:144]: Actually, that sounds better to me. I'm gonna make it throw.

**Review by davidfowl — APPROVED** [review]

**Comment by codeputer** [comment]
Awesome...can we pull this code via the update?

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by JunTaoLuo** [comment]
@codeputer Yes, you can get this change by installing our nightly, see https://github.com/dotnet/tye/blob/master/docs/getting_started.md#working-with-ci-builds

**Comment by codeputer** [comment]
Thanks… I’m still getting this error on the microservice program…

This it the directory structure… I need to review/understand the dockerfile and why the copies are not working correctly – hints on where to look are appreciated.
[cid:image002.jpg@01D6A21E.FFB35630]

**Comment by GeorgeS2019** [comment]
I still have problem getting[ the CI version](https://github.com/dotnet/tye/blob/master/docs/getting_started.md#working-with-ci-builds) to work with the  .NET5 (not .NETCore 3.1) version

**Pull request merged**